Gems/ROS2/Code/Tests/UrdfParserTest.cpp

```diff
@@ -0,0 +1,453 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/std/string/string.h>
+#include <AzTest/AzTest.h>
+#include <RobotImporter/URDF/UrdfParser.h>
+#include <RobotImporter/Utils/RobotImporterUtils.h>
+
+namespace UnitTest
+{
+
+    class UrdfParserTest : public AllocatorsTestFixture
```

Comment:
repeat: This code will need to be updated to use [LeakDetectionFixture](https://github.com/o3de/o3de/commit/05c122feab34b37cab4e8df3fac7be14c6f3029f) for it to compile.

Reply:
Done